File: 2554. Maximum Number of Integers to Choose From a Range.py
class Solution:
    def maxCount(self, banned: List[int], n: int, maxSum: int) -> int:
        banset = set(banned)
        sum=0
        count=0

        for i in range(1, n+1):
            if i not in banset and sum+i<=maxSum:
                sum+=i
                count+=1
        return count
        

File: 2554. Maximum Number of Integers to Choose From a Range.js
/**
 * @param {number[]} banned
 * @param {number} n
 * @param {number} maxSum
 * @return {number}
 */
var maxCount = function (banned, n, maxSum) {
  const banset = new Set(banned);
  let sum = 0;
  let count = 0;

  for (let i = 1; i <= n; i++) {
    if (!banset.has(i) && sum + i <= maxSum) {
      sum += i;
      count++;
    }
  }
  return count;
};
